src/python/pants/backend/java/dependency_inference/java_parser.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import json
+import logging
+import os.path
+from dataclasses import dataclass
+
+from pants.backend.java.dependency_inference.java_parser_launcher import (
+    JavaParserCompiledClassfiles,
+    java_parser_artifact_requirements,
+)
+from pants.backend.java.dependency_inference.types import JavaSourceDependencyAnalysis
+from pants.core.util_rules.source_files import SourceFiles
+from pants.engine.fs import AddPrefix, Digest, DigestContents, MergeDigests
+from pants.engine.process import BashBinary, FallibleProcessResult, Process, ProcessExecutionFailure
+from pants.engine.rules import Get, collect_rules, rule
+from pants.jvm.resolve.coursier_fetch import MaterializedClasspath, MaterializedClasspathRequest
+from pants.jvm.resolve.coursier_setup import Coursier
+from pants.util.logging import LogLevel
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class FallibleJavaSourceDependencyAnalysisResult:
+    process_result: FallibleProcessResult
+
+
+@rule(level=LogLevel.DEBUG)
+async def resolve_fallible_result_to_analysis(
+    fallible_result: FallibleJavaSourceDependencyAnalysisResult,
+) -> JavaSourceDependencyAnalysis:
+    # For whatever reason this doesn't work despite there being a rule that ostensibly
+    # does this conversion automatically.
+    # result = await Get(ProcessResult, FallibleProcessResult, fallible_result.process_result)
```

Comment:
Is there an open issue for this?

Reply:
Yeah apparently this should be resolved here: https://github.com/pantsbuild/pants/pull/12725